core/chains/evm/gas/block_history_estimator.go

```diff
@@ -838,20 +838,23 @@ func (b *BlockHistoryEstimator) EffectiveGasPrice(block evmtypes.Block, tx evmty
 			b.logger.Warnw("Got transaction type 0x2 but one of the required EIP1559 fields was missing, falling back to gasPrice", "block", block, "tx", tx)
 			return tx.GasPrice
 		}
-		if tx.MaxFeePerGas.Cmp(block.BaseFeePerGas) < 0 {
-			// This should not pass config validation
+		assumptionViolation := false
```

Comment:
Did you consider moving this up here instead?
```suggestion
		if tx.GasPrice != nil {
			// Always use the gas price if provided
			return tx.GasPrice
		}
```
Or are we trying to retain the logs?

Reply:
Trying to retain the logs 👍

Reply:
If we don't think the log lines will be useful, dropping them will make the logic a good bit simpler (as you suggested !) - but I think it'll come in handy.

Reply:
I'm not sure if they are still important but there are sort of two different cases now, and warn level might be more appropriate when we just use gas price anyways.

Reply:
I went back and forth on that one as well. The current `AssumptionViolationw` logs at the Error level. 

I can create an AssumptionViolation logger at the warn level instead to keep the `AssumptionViolation` prefix.

Reply:
IMHO this is actually a confusing helper that we should phase out. I'm not sure how best to report these conditions though.

Reply:
Maybe [here](https://github.com/smartcontractkit/chainlink/blob/587b7131c77dc222af1c40e966a77721d1c6b323/core/chains/evm/gas/block_history_estimator.go#L755) ?